### PR TITLE
Fixes for elements 21 and 22

### LINKED ISSUE #4: Needs to create default wallet and createrawtransaction arguments have changed
Elements now requires a default wallet creating and I think the createrawtransaction interface has also changed, causing error.

To fix the no default asset error, `createasset.sh` could call:

```
# Create default wallets
${ELEMENTSPATH}/elements-cli -conf=$C1 createwallet ""
${ELEMENTSPATH}/elements-cli -conf=$C1 rescanblockchain
${ELEMENTSPATH}/elements-cli -conf=$C2 createwallet ""
${ELEMENTSPATH}/elements-cli -conf=$C2 rescanblockchain
```
This worked for me but I couldn't commit the fix as I later encountered an error running the following line from the [regtest example](https://github.com/Blockstream/liquid-swap/blob/master/docs/regtest-example.md):

```
liquidswap-cli --regtest -c $C1 propose $ASSET1 1 $ASSET2 2 --output proposal.txt

liquidswap ERROR createrawtransaction [{"txid":"hex","vout":n,"sequence":n,"pegin_bitcoin_tx":"hex","pegin_txout_proof":"hex","pegin_claim_script":"hex"},...] [{"address":amount,"asset":"str"},{"data":"hex"},{"vdata":"hex"},{"burn":"hex"},{"fee":amount},...] ( locktime replaceable )
```
I think this is because the interface to `createrawtransaction` has changed - not sure. Everything I ran up to that point worked ok (after fixing the no default wallet issue).

### PR DESCRIPTION
**Pull request by andreabonel**
Fixes for elements 21: 
- wallet creation
- createrawtransation (new format)

Fix for elements 22:
- decoderawtransaction (new format)

Fixes: https://github.com/Blockstream/liquid-swap/issues/4 and https://github.com/Blockstream/liquid-swap/issues/5

### CODE REVIEW

**Comment by wintercooled** [comment]
Tested ACK:

```
git clone https://github.com/andreabonel/liquid-swap.git
cd liquid-swap

git checkout fixes-for-elements-21-and-22 

pip install .[GUI,CLI]

. ./tools/set_env.sh $HOME/elements-0.22.0.2/bin

./tools/start_liquid_instances.sh

$ e1c getnewaddress
el1qqt6y5eywtaxchr8gx2g3xau7my8dvytvze50cnx6e82mwd86g6cmchaluvgp048nrvs9276m8slyel4zl52kwzwlqd70ec0va

$ e1c generatetoaddress 1 el1qqt6y5eywtaxchr8gx2g3xau7my8dvytvze50cnx6e82mwd86g6cmchaluvgp048nrvs9276m8slyel4zl52kwzwlqd70ec0va
[
  "9464f6f6d8f22634f2c75c7d4515ff357175c177fa5f5f7b0d9d1fd1e876bfbe"
]

$ e1c getbalance
{
  "bitcoin": 21000000.00000000
}

. ./tools/createassets.sh

$ e1c generatetoaddress 1 el1qqt6y5eywtaxchr8gx2g3xau7my8dvytvze50cnx6e82mwd86g6cmchaluvgp048nrvs9276m8slyel4zl52kwzwlqd70ec0va
[
  "9464f6f6d8f22634f2c75c7d4515ff357175c177fa5f5f7b0d9d1fd1e876bfbe"
]

$ e1c getbalance
{
  "bitcoin": 20999899.99999371,
  "0ea147786fc5e3db56af56306fbf8acae4c91ba7828a68a352a8723b326152cd": 10.00000000
}

$ e2c getbalance
{
  "bitcoin": 99.99999636,
  "7e67ff31e4d26f88fed62525442d35d343f6cbb715a19208a4b0d7e2437eb646": 20.00000000
}

liquidswap-cli --regtest -c $C1 propose $ASSET1 1 $ASSET2 2 --output proposal.txt
liquidswap-cli --regtest -c $C2 info proposal.txt
liquidswap-cli --regtest -c $C2 accept proposal.txt --output accepted.txt
liquidswap-cli --regtest -c $C1 info accepted.txt
liquidswap-cli --regtest -c $C1 finalize accepted.txt --send

e1c generatetoaddress 1 el1qqt6y5eywtaxchr8gx2g3xau7my8dvytvze50cnx6e82mwd86g6cmchaluvgp048nrvs9276m8slyel4zl52kwzwlqd70ec0va
[
  "13f8eeadcf05834e374c81d8d8754a5e5c334f90779bf3bdeb7067b5fb2352e6"
]

e1c getbalance
{
  "bitcoin": 20999899.99998995,
  "7e67ff31e4d26f88fed62525442d35d343f6cbb715a19208a4b0d7e2437eb646": 2.00000000,
  "0ea147786fc5e3db56af56306fbf8acae4c91ba7828a68a352a8723b326152cd": 9.00000000
}

e2c getbalance
{
  "bitcoin": 99.99999260,
  "7e67ff31e4d26f88fed62525442d35d343f6cbb715a19208a4b0d7e2437eb646": 18.00000000,
  "0ea147786fc5e3db56af56306fbf8acae4c91ba7828a68a352a8723b326152cd": 1.00000000
}

./tools/stop_liquid_instances.sh
```

File: liquidswap/swap.py [hunk 1]
```diff
@@ -343,13 +369,23 @@ def accept(tx_p,
...
+        tx = connection.createrawtransaction(
+            inputs,
+            sort_dict(values2btc(map_amount, None)),
+            NLOCKTIME,
+            IS_REPLACEABLE,
+            sort_dict(map_asset)
+        )
+    else:
+        map_asset.update(map_asset_p)
+        map_asset.update(map_asset_r)
```
Comment (LeoComandini) [inline liquidswap/swap.py]: this 2 lines should not be necessary, we already update `map_asset` few lines above

File: liquidswap/swap.py [hunk 2]
```diff
@@ -261,16 +278,25 @@ def accept(tx_p,
...
+            [],
+            {c_address_p: sat2btc(amount_r)},
+            NLOCKTIME,
+            IS_REPLACEABLE,
+            {c_address_p: asset_r}
+        )
+    else:
+        txu = connection.createrawtransaction(
+            [],
+            [{c_address_p: sat2btc(amount_r),"asset": asset_r}],
```
Comment (LeoComandini) [inline liquidswap/swap.py]: nit
```suggestion
            [{c_address_p: sat2btc(amount_r), "asset": asset_r}],
```

File: liquidswap/swap.py [hunk 3]
```diff
@@ -127,13 +136,21 @@ def propose(amount_p, asset_p,
...
+        tx = connection.createrawtransaction(
+            inputs,
+            values2btc(map_amount, None),
+            NLOCKTIME,
+            IS_REPLACEABLE,
+            map_asset
+        )
+    else:
+        tx = connection.createrawtransaction(
+            inputs,  # [{DUMMY_ADDRESS_CONFIDENTIAL[is_mainnet]: sat2btc(amount_p), "asset": asset_p}],
```
Comment (LeoComandini) [inline liquidswap/swap.py]: nit: move comment to below to its line or remove it

File: liquidswap/swap.py [hunk 4]
```diff
@@ -58,14 +58,23 @@ def propose(amount_p, asset_p,
...
+            [],
+            {DUMMY_ADDRESS_CONFIDENTIAL[is_mainnet]: sat2btc(amount_p)},
+            NLOCKTIME,
+            IS_REPLACEABLE,
+            {DUMMY_ADDRESS_CONFIDENTIAL[is_mainnet]: asset_p}
+        )
+    else:
+        txu = connection.createrawtransaction(
+            [],
+            [{DUMMY_ADDRESS_CONFIDENTIAL[is_mainnet]: sat2btc(amount_p), "asset": asset_p}],
```
Comment (LeoComandini) [inline liquidswap/swap.py]: super-nit: use `'` instead of `"` for consistency (but feel free to ignore)

**Comment by LeoComandini** [comment]
utACK 47e9941137a9111d39792dcc3c11527f90a3048b, with comments

**Comment by LeoComandini** [comment]
utACK 14d459af60ad1de9d611b7a0ec26980f9dda2b03

**Comment by LeoComandini** [comment]
Thanks @andreabonel !

**Pull request merged**